client-runtime/src/main/java/com/microsoft/rest/v2/http/HttpClient.java

```diff
@@ -6,14 +6,16 @@
 
 package com.microsoft.rest.v2.http;
 
+import com.microsoft.rest.v2.policy.RequestPolicy;
 import rx.Single;
+import rx.functions.Func1;
 
 import java.io.IOException;
 
 /**
  * A generic interface for sending HTTP requests and getting responses.
  */
-public abstract class HttpClient {
+public abstract class HttpClient implements RequestPolicy {
```

Comment:
I'm a little wary about this. I'm not entirely certain what RequestPolicy is supposed to be, but I don't think that HttpClient should implement it. This feels like maybe the name "RequestPolicy" isn't right.